tests/integration/goldens/asset/google/cloud/asset_v1/services/asset_service/transports/grpc_asyncio.py

```diff
@@ -1065,71 +1066,80 @@ def _prep_wrapped_messages(self, client_info):
                 self.query_assets,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.create_saved_query: gapic_v1.method_async.wrap_method(
+            self.create_saved_query: self._wrap_method_with_kind(
                 self.create_saved_query,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.get_saved_query: gapic_v1.method_async.wrap_method(
+            self.get_saved_query: self._wrap_method_with_kind(
                 self.get_saved_query,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.list_saved_queries: gapic_v1.method_async.wrap_method(
+            self.list_saved_queries: self._wrap_method_with_kind(
                 self.list_saved_queries,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.update_saved_query: gapic_v1.method_async.wrap_method(
+            self.update_saved_query: self._wrap_method_with_kind(
                 self.update_saved_query,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.delete_saved_query: gapic_v1.method_async.wrap_method(
+            self.delete_saved_query: self._wrap_method_with_kind(
                 self.delete_saved_query,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.batch_get_effective_iam_policies: gapic_v1.method_async.wrap_method(
+            self.batch_get_effective_iam_policies: self._wrap_method_with_kind(
                 self.batch_get_effective_iam_policies,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.analyze_org_policies: gapic_v1.method_async.wrap_method(
+            self.analyze_org_policies: self._wrap_method_with_kind(
                 self.analyze_org_policies,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.analyze_org_policy_governed_containers: gapic_v1.method_async.wrap_method(
+            self.analyze_org_policy_governed_containers: self._wrap_method_with_kind(
                 self.analyze_org_policy_governed_containers,
                 default_timeout=None,
                 client_info=client_info,
             ),
-            self.analyze_org_policy_governed_assets: gapic_v1.method_async.wrap_method(
+            self.analyze_org_policy_governed_assets: self._wrap_method_with_kind(
                 self.analyze_org_policy_governed_assets,
                 default_timeout=None,
                 client_info=client_info,
             ),
-         }
+        }
+
+    def _wrap_method_with_kind(self, func, *args, **kwargs):
+        if "kind" in inspect.signature(gapic_v1.method_async.wrap_method).parameters:  # pragma: NO COVER
```

Comment:
@vchudnov-g 
Will address 1. and 3.

For 2. I'm a little confused about what the ask is.

- The `wrap_method` for a sync call is different for an async call and let's discuss only the latter here (since we're not touching the sync code path.
- The default value is `grpc_asyncio` and shouldn't be `grpc` for the async `wrap_method`. See this comment: https://github.com/googleapis/python-api-core/pull/688#discussion_r1735300998.
- This check is added to not break old users. I think for new code, it should be a requirement to install the latest version of `google-api-core`.

Reply:
So if we require the latest version of `google-api-core`, then the `gapic_v1.method_async.wrap_method` will always take `kind`, right? So why do we need to check that `kind` is included in its parameter list? Why not add `kind` unconditionally? That's what's throwing me off.

Reply:
> So if we require the latest version of google-api-core, then the gapic_v1.method_async.wrap_method will always take kind, right? So why do we need to check that kind is included in its parameter list? Why not add kind unconditionally? That's what's throwing me off.

To not break someone using asynchronous gRPC with an older version of `google-api-core`.

> We'll require the latest version for asynchronous REST.

Reply:
To be more explicit, I don't think we need to bump the minimum version of `google-api-core`.

Just within the asynchronous REST transport, we can raise an exception (since that feature will depend on the latest version of core).

For gRPC, we need to guard it within this condition so that if someone is still using an older version of api-core, we don't break them.

Reply:
OK, so our dependency story is a bit complex then, right?:
1. we won't require a newer version of api-core to install the new GAPICs
2. at run-time, we'll error on async rest if the version of api-core is not upgraded to to support async rest.

When do we plan to change #1, and require a newer version of api-core for installation? After we complete implementation of the entire async rest work (i.e. beyond just M1)?

Reply:
You're right about 1. and 2.

> When do we plan to change https://github.com/googleapis/gapic-generator-python/pull/1, and require a newer version of api-core for installation? After we complete implementation of the entire async rest work (i.e. beyond just M1)?

I'm not sure about this honestly. But whenever we do that, we should remove these checks and I've added a note for that. I've previously noticed that we guard any new changes from core with a version comparison rather than bumping the minimum version right away.

Reply:
Could you file an issue to see when we we can require a newer version? Then in the places where we have these checks, let's add a TODO referencing that issue. That way, when we get to the issue and update the version requirement, we can find the places with the checks to remove by just searching for the issue # in the code.

Reply:
Also, we just chatted by a separate channel: at the moment, this code is only called for async grpc, so the behavior doesn't change. We will be calling it for async rest as well in an upcoming PR.